### PR TITLE
sorbet: set utils/inreplace.rb to true

### PR DESCRIPTION
**Pull request by vidusheeamoli**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
This PR moves `utils/inreplace.rb` to true in `sorbet/files.yaml` and deals with the resulting type errors.

### CODE REVIEW

File: Library/Homebrew/utils/inreplace.rb [hunk 1]
```diff
@@ -25,13 +25,13 @@ def inreplace(paths, before = nil, after = nil, audit_result = true)
...
       Array(paths).each do |path|
-        s = File.open(path, "rb", &:read).extend(StringInreplaceExtension)
+        s = (File.open(path, "rb", &:read) || "").extend(StringInreplaceExtension)
 
         if before.nil? && after.nil?
           yield s
         else
           after = after.to_s if after.is_a? Symbol
-          s.gsub!(before, after, audit_result)
+          s.gsub!(T.must(before), after, audit_result)
```
Comment (vidusheeamoli) [inline Library/Homebrew/utils/inreplace.rb:34]: This would not be needed once we resolve the issue with the `gsub!` method.

**Comment by issyl0** [comment]
This also needs a rebase for the same reason as #8019, please. 😩

**Comment by issyl0** [comment]
The tests are failing. 😢 I'll have some time to look into it in more detail tomorrow, but I think overall I'm going to leave this review to @mistydemeo as she had the original vision for this refactoring!

**Comment by mistydemeo** [comment]
This is a good start! I'd like you to think a bit more closely about how the code is structured today and the changes you're looking to make. Here's a few questions to think about:

* Why is most of the code in a mixin today? (Hint: it's adding methods to other objects.) Do we still need that code to be a mixin when you're done refactoring?
* Which objects are you passing around that need to expose a `gsub!` method? What type is that today, and what type will that be after this PR?

File: Library/Homebrew/utils/inreplace.rb [hunk 2]
```diff
@@ -25,15 +35,17 @@ def inreplace(paths, before = nil, after = nil, audit_result = true)
       errors["`paths` (first) parameter"] = ["`paths` was empty"] if paths.blank?
 
       Array(paths).each do |path|
-        s = File.open(path, "rb", &:read).extend(StringInreplaceExtension)
+        s = (File.open(path, "rb", &:read) || "").extend(StringInreplaceExtension)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/inreplace.rb:38]: This is a bit hard to follow, could it be two variable declarations instead of one?

File: Library/Homebrew/sorbet/rbi/utils/inreplace.rbi [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+class StringInreplaceExtension
+  sig { params(before: String, after: String).returns(T.nilable(String)) }
+  def sub!(before, after)
+  end
+
+  sig { params(before: T.nilable(String), after: T.nilable(String), audit_result: T::Boolean).returns(T.nilable(String)) }
+  def gsub!(before, after, audit_result = true); end
+
+  def change_make_var!(flag, new_value)
```
Comment (mistydemeo) [inline Library/Homebrew/sorbet/rbi/utils/inreplace.rbi:19]: Do we want to add types to these three methods?

File: Library/Homebrew/dev-cmd/bottle.rb [hunk 4]
```diff
@@ -479,7 +479,8 @@ def merge
         path = Pathname.new((HOMEBREW_REPOSITORY/bottle_hash["formula"]["path"]).to_s)
         update_or_add = nil
 
-        Utils::Inreplace.inreplace(path) do |s|
+        Utils::Inreplace.inreplace(path) do |str|
+          s = StringInreplaceExtension.new(str)
```
Comment (mistydemeo) [inline Library/Homebrew/dev-cmd/bottle.rb:483]: I think `str` is already a `StringInreplaceExtension` here.

File: Library/Homebrew/sorbet/rbi/utils/inreplace.rbi [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  def inreplace(paths, before = nil, after = nil, audit_result = true); end
+
+end
+
+class StringInreplaceExtension
+  sig { params(before: String, after: String).returns(T.nilable(String)) }
+  def sub!(before, after)
+  end
+
+  sig { params(before: T.nilable(String), after: T.nilable(String), audit_result: T::Boolean).returns(T.nilable(String)) }
```
Comment (reitermarkus) [inline Library/Homebrew/sorbet/rbi/utils/inreplace.rbi:16]: Why is this `nilable` for `gsub!`?

File: Library/Homebrew/sorbet/rbi/utils/inreplace.rbi [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+module Utils::Inreplace
+  include Kernel
+
+  sig { params(paths: T::Array[T.untyped], before: T.nilable(String), after: T.nilable(String), audit_result: T::Boolean).void }
+  def inreplace(paths, before = nil, after = nil, audit_result = true); end
+
+end
+
+class StringInreplaceExtension
+  sig { params(before: String, after: String).returns(T.nilable(String)) }
```
Comment (reitermarkus) [inline Library/Homebrew/sorbet/rbi/utils/inreplace.rbi:12]: I think `before` can also be a `Regexp`.
